cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -727,6 +793,13 @@ TEST_F(GcsIntegrationTest, CreateDirUri) {
   ASSERT_RAISES(Invalid, fs->CreateDir("gs://" + RandomBucketName(), true));
 }
 
+TEST_F(GcsIntegrationTest, DeleteBucketDirSuccess) {
+  auto fs = GcsFileSystem::Make(TestGcsOptions());
+  ASSERT_OK(fs->CreateDir("pyarrow-filesystem/", /*recursive=*/true));
+  EXPECT_FALSE(fs->CreateDir("/", false).ok());
```

Comment:
Use `ASSERT_RAISES` with the expected error here.

Reply:
done.